Source/Core/InputCommon/ControllerInterface/Xlib/XInput2.cpp

```diff
@@ -3,10 +3,8 @@
 
 #include "InputCommon/ControllerInterface/Xlib/XInput2.h"
 
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wregister"
 #include <X11/XKBlib.h>
-#pragma GCC diagnostic pop
```

Comment:
Is this change consistent across all X11/XKBlib.h headers dolphin might run into? Or just the version you have installed?

Reply:
I was able to track down the commit which removed the register keyword from \<X11/XKBlib.h\>, but I know nothing of the minimum X11 version required.  https://github.com/freedesktop/xorg-libX11/commit/6dd618e5ea8e7558dd8e99a2777ab7b69211789c
Moreover, I could not even re-introduce the warning by modifying my copy of \<X11/XKBlib.h\>.  Looking into things, it seems we used to access X11 in our CMakeLists.txt with an `include_directories` command, but this was later changed.
https://github.com/dolphin-emu/dolphin/blob/9b201815f257892c0c5beb405c0ff0af7cd432dc/CMakeLists.txt#L450
Since we now access it as a system header, no warnings are emitted regardless.  So while it's not for the reasons I thought, removing this warning should be fine.